src/airobot/robot/ur5e_pybullet.py

```diff
@@ -0,0 +1,736 @@
+        """
+        Return the end effector pose
+
+        Returns:
+            end effector position, quaternion,
+            rotation matrix, euler angles
+        """
+        info = p.getLinkState(self.robot_id, self.ee_link_id)
+        pos = info[4]
+        quat = info[5]
+        rot_mat = p.getMatrixFromQuaternion(quat)
+        euler = p.getEulerFromQuaternion(quat)  # [roll, pitch, yaw]
+        return list(pos), list(quat), list(rot_mat), list(euler)
+
+    def get_ee_force(self):
+        """
+        TODO add force sensor on the end effector
+        """
+        raise NotImplementedError
+
+    def compute_ik(self, pos, ori=None):
+        """
+        Compute the inverse kinematics solution given the
+        position and orientation of the end effector
+
+        Args:
+            pos (list): position
+            ori (list): orientation. It can be euler angles
+                (roll, pitch, yaw) or quaternion.
+
+        Returns:
+            inverse kinematics solution (joint angles, list)
+        """
+        if ori is not None:
+            if len(ori) == 3:
+                # [roll, pitch, yaw]
+                ori = p.getQuaternionFromEuler(ori)
+            if len(ori) != 4:
+                raise ValueError('Orientation should be either '
+                                 'euler angles or quaternion')
+            jnt_poss = p.calculateInverseKinematics(self.robot_id,
+                                                    self.ee_link_id,
+                                                    pos,
+                                                    ori,
+                                                    jointDamping=self._ik_jds)
+        else:
+            jnt_poss = p.calculateInverseKinematics(self.robot_id,
+                                                    self.ee_link_id,
+                                                    pos,
+                                                    jointDamping=self._ik_jds)
+        jnt_poss = list(jnt_poss)
+        return jnt_poss[:len(self.actuator_ids)]
+
+    def _mimic_gripper(self, joint_val):
+        """
+        Given the value for the first joint,
+        mimic the joint values for the rest joints
+        """
+        jnt_vals = [joint_val]
+        for i in range(1, len(self.gripper_jnt_names)):
+            jnt_vals.append(joint_val * self._gripper_mimic_coeff[i])
+        return jnt_vals
+
+    def _wait_to_reach_jnt_goal(self, goal, joint_name=None, mode='pos'):
+        """
+        Block the code to wait for the joint moving to the specified goal.
+        The goal can be a desired velocity(s) or a desired position(s).
+        Max waiting time is self.cfgs.TIMEOUT_LIMIT
+
+        Args:
+            goal (float or list): goal positions or velocities
+            joint_name (str): if it's none, all the actuated
+                joints are compared.
+                Otherwise, only the specified joint is compared
+            mode (str): 'pose' or 'vel'
```

Comment:
docstring typo, mode (str): 'pose' >>> mode (str): 'pos'

Reply:
thanks. It has been corrected.